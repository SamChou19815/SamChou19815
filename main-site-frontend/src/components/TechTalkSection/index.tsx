import React, { ReactElement } from 'react';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Icon from '@material-ui/core/Icon';
import styles from './index.module.css';
import ButtonLink from '../Common/ButtonLink';
import ConsoleSection from '../Common/ConsoleSection';

const schoolIcon = <Icon>school</Icon>;

export default (): ReactElement => (
  <ConsoleSection id="tech-talks" title="./tech-talks -all">
    <div className={styles.TechTalkContainer}>
      <Card className={styles.TechTalkCard}>
        <CardHeader avatar={schoolIcon} title="How to scale" subheader="Learning Series" />
        <CardContent>
          {'Tips on scaling your codebase and your workload,'}
          {" with lessons learned from Samwise's codebase."}
        </CardContent>
        <CardActions>
          <ButtonLink href="/how-to-scale.pdf" openInNewTab>
            Slides
          </ButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.TechTalkCard}>
        <CardHeader avatar={schoolIcon} title="Intro to Firebase" subheader="DevSesh" />
        <CardContent>
          Tech stack discussion on Firebase, and why Samwise switched to Firebase.
        </CardContent>
        <CardActions>
          <ButtonLink href="/intro-to-firebase.pdf" openInNewTab>
            Slides
          </ButtonLink>
          <ButtonLink href="https://jessicahong9.github.io/" openInNewTab>
            {"Co-speaker Jessica's website"}
          </ButtonLink>
        </CardActions>
      </Card>
    </div>
  </ConsoleSection>
);
