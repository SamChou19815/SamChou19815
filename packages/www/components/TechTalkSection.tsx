import React, { ReactElement } from 'react';

import classnames from 'classnames';

import techTalks from '../data/tech-talks';
import CardHeader from './Common/CardHeader';
import ConsoleSection from './Common/ConsoleSection';
import styles from './TechTalkSection.module.css';

const TechTalkSection = (): ReactElement => (
  <ConsoleSection id="tech-talks" title="dev-sam tech-talks">
    <div className={styles.TechTalkContainer}>
      {techTalks.map(({ title, type, description, link }) => (
        <div key={title} className={classnames('card', styles.TechTalkCard)}>
          <CardHeader title={title} subheader={type} />
          <div className="card__body">{description}</div>
          <div className="card__footer">
            <a href={link} className="button button--link">
              SLIDES
            </a>
          </div>
        </div>
      ))}
    </div>
  </ConsoleSection>
);

export default TechTalkSection;
