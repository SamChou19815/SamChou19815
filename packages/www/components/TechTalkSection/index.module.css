.TechTalkContainer {
  display: flex;
  justify-content: center;
  align-items: flex-start;
  align-content: flex-start;
  flex-direction: row;
  flex-wrap: wrap;
}

@media screen and (max-width: 800px) {
  .TechTalkCard {
    margin: 1em auto;
    width: 90vw;
  }
}

@media screen and (min-width: 800px) and (max-width: 1280px) {
  .TechTalkCard {
    margin: 1em;
    width: 450px;
  }
}

@media screen and (min-width: 1280px) {
  .TechTalkCard {
    margin: 1em;
    width: 300px;
  }
}
