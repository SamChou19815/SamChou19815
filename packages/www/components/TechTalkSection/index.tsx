import React, { ReactElement } from 'react';

import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';

import ConsoleSection from '../Common/ConsoleSection';
import styles from './index.module.css';

import MaterialButtonLink from 'lib-react/MaterialButtonLink';

export default (): ReactElement => (
  <ConsoleSection id="tech-talks" title="tech-talks --all">
    <div className={styles.TechTalkContainer}>
      <Card className={styles.TechTalkCard}>
        <CardHeader title="How to scale" subheader="Learning Series" />
        <CardContent>
          Tips on scaling your codebase and your workload, with lessons learned from Samwise&apos;s
          codebase.
        </CardContent>
        <CardActions>
          <MaterialButtonLink href="/how-to-scale.pdf">Slides</MaterialButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.TechTalkCard}>
        <CardHeader title="Intro to Firebase" subheader="DevSesh" />
        <CardContent>
          Tech stack discussion on Firebase, and why Samwise switched to Firebase.
        </CardContent>
        <CardActions>
          <MaterialButtonLink href="/intro-to-firebase.pdf">Slides</MaterialButtonLink>
          <MaterialButtonLink href="https://jessicahong9.github.io/">
            Co-speaker&apos;s website
          </MaterialButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.TechTalkCard}>
        <CardHeader title="Build your programming language" subheader="DevSesh" />
        <CardContent>
          A tutorial of making a simple programming language derived from lambda-calculus.
        </CardContent>
        <CardActions>
          <MaterialButtonLink href="/build-your-own-programming-language.pdf">
            Slides
          </MaterialButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.TechTalkCard}>
        <CardHeader title="Build a (simplified) React" subheader="DevSesh" />
        <CardContent>
          A tutorial of making a simplified React runtime with support for useState and useEffect
          hooks.
        </CardContent>
        <CardActions>
          <MaterialButtonLink href="/build-simplified-react.pdf">Slides</MaterialButtonLink>
        </CardActions>
      </Card>
    </div>
  </ConsoleSection>
);
