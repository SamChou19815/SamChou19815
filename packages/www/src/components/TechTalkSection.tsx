import React, { ReactElement } from 'react';

import DATASET_TECH_TALKS from '../data/tech-talks';
import CardHeader from './Common/CardHeader';

const TechTalkSection = (): ReactElement => (
  <div className="card-container">
    {DATASET_TECH_TALKS.map(({ title, description, link }) => (
      <div key={title} className="card responsive-card">
        <CardHeader title={title} />
        <div className="card__body">{description}</div>
        <div className="card__footer">
          <a href={link} className="button button--link">
            SLIDES
          </a>
        </div>
      </div>
    ))}
  </div>
);

export default TechTalkSection;
