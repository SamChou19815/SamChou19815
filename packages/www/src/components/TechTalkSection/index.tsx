import React, { ReactElement } from 'react';

import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';
import School from '@material-ui/icons/School';
import MaterialButtonLink from 'lib-react/MaterialButtonLink';

import ConsoleSection from '../Common/ConsoleSection';
import styles from './index.module.css';

const schoolIcon = <School />;

export default (): ReactElement => (
  <ConsoleSection id="tech-talks" title="tech-talks --all">
    <div className={styles.TechTalkContainer}>
      <Card className={styles.TechTalkCard}>
        <CardHeader avatar={schoolIcon} title="How to scale" subheader="Learning Series" />
        <CardContent>
          Tips on scaling your codebase and your workload, with lessons learned from Samwise&apos;s
          codebase.
        </CardContent>
        <CardActions>
          <MaterialButtonLink href="/how-to-scale.pdf">Slides</MaterialButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.TechTalkCard}>
        <CardHeader avatar={schoolIcon} title="Intro to Firebase" subheader="DevSesh" />
        <CardContent>
          Tech stack discussion on Firebase, and why Samwise switched to Firebase.
        </CardContent>
        <CardActions>
          <MaterialButtonLink href="/intro-to-firebase.pdf">Slides</MaterialButtonLink>
          <MaterialButtonLink href="https://jessicahong9.github.io/">
            Co-speaker Jessica&apos;s website
          </MaterialButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.TechTalkCard}>
        <CardHeader
          avatar={schoolIcon}
          title="How to build your own programming language"
          subheader="DevSesh"
        />
        <CardContent>
          A tutorial of making a simple programming language derived from lambda-calculus.
        </CardContent>
        <CardActions>
          <MaterialButtonLink href="/build-your-own-programming-language.pdf">
            Slides
          </MaterialButtonLink>
        </CardActions>
      </Card>
    </div>
  </ConsoleSection>
);
