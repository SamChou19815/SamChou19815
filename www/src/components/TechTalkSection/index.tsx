import React, { ReactElement } from 'react';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import School from '@material-ui/icons/School';
import MaterialButtonLink from 'sam-react-common/MaterialButtonLink';
import styles from './index.module.css';
import ConsoleSection from '../Common/ConsoleSection';

const schoolIcon = <School />;

export default (): ReactElement => (
  <ConsoleSection id="tech-talks" title="./tech-talks -all">
    <div className={styles.TechTalkContainer}>
      <Card className={styles.TechTalkCard}>
        <CardHeader avatar={schoolIcon} title="How to scale" subheader="Learning Series" />
        <CardContent>
          Tips on scaling your codebase and your workload, with lessons learned from Samwise&apos;s
          codebase.
        </CardContent>
        <CardActions>
          <MaterialButtonLink href="/how-to-scale.pdf">Slides</MaterialButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.TechTalkCard}>
        <CardHeader avatar={schoolIcon} title="Intro to Firebase" subheader="DevSesh" />
        <CardContent>
          Tech stack discussion on Firebase, and why Samwise switched to Firebase.
        </CardContent>
        <CardActions>
          <MaterialButtonLink href="/intro-to-firebase.pdf">Slides</MaterialButtonLink>
          <MaterialButtonLink href="https://jessicahong9.github.io/">
            Co-speaker Jessica&apos;s website
          </MaterialButtonLink>
        </CardActions>
      </Card>
    </div>
  </ConsoleSection>
);
