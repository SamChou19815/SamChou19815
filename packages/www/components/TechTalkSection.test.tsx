import React from 'react';

import renderer from 'react-test-renderer';

import TechTalkSection from './TechTalkSection';

it('TechTalkSection matches snapshot.', () => {
  const tree = renderer.create(<TechTalkSection />).toJSON();
  expect(tree).toMatchSnapshot();
});
