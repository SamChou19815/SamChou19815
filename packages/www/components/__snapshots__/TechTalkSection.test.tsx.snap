// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechTalkSection matches snapshot. 1`] = `
<section
  className="Section"
  id="tech-talks"
>
  <h3
    className="Title"
  >
    <code>
      $Â 
      dev-sam tech-talks
    </code>
  </h3>
  <div
    className="TechTalkContainer"
  >
    <div
      className="MuiPaper-root MuiCard-root TechTalkCard MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            How to scale
          </span>
          <span
            className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            Learning Series
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root"
      >
        Tips on scaling your codebase and your workload, with lessons learned from Samwise's codebase.
      </div>
      <div
        className="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="/how-to-scale.pdf"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Slides
            </a>
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiCard-root TechTalkCard MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Intro to Firebase
          </span>
          <span
            className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            DevSesh
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root"
      >
        Tech stack discussion on Firebase, and why Samwise switched to Firebase.
      </div>
      <div
        className="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="/intro-to-firebase.pdf"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Slides
            </a>
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="https://jessicahong9.github.io/"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Co-speaker's website
            </a>
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiCard-root TechTalkCard MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Build your programming language
          </span>
          <span
            className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            DevSesh
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root"
      >
        A tutorial of making a simple programming language derived from lambda-calculus.
      </div>
      <div
        className="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="/build-your-own-programming-language.pdf"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Slides
            </a>
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiCard-root TechTalkCard MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Build a (simplified) React
          </span>
          <span
            className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            DevSesh
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root"
      >
        A tutorial of making a simplified React runtime with support for useState and useEffect hooks.
      </div>
      <div
        className="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="/build-simplified-react.pdf"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Slides
            </a>
          </span>
        </button>
      </div>
    </div>
  </div>
</section>
`;
